phpBB/phpbb/install/module/update_database/task/update.php

```diff
@@ -0,0 +1,207 @@
+use phpbb\install\exception\user_interaction_required_exception;
+use phpbb\install\task_base;
+
+/**
+ * Database updater task
+ */
+class update extends task_base
+{
+	/**
+	 * @var \phpbb\cache\driver\driver_interface
+	 */
+	protected $cache;
+
+	/**
+	 * @var \phpbb\config\config
+	 */
+	protected $config;
+
+	/**
+	 * @var \phpbb\extension\manager
+	 */
+	protected $extension_manager;
+
+	/**
+	 * @var \phpbb\filesystem\filesystem
+	 */
+	protected $filesystem;
+
+	/**
+	 * @var \phpbb\install\helper\config
+	 */
+	protected $installer_config;
+
+	/**
+	 * @var \phpbb\install\helper\iohandler\iohandler_interface
+	 */
+	protected $iohandler;
+
+	/**
+	 * @var \phpbb\language\language
+	 */
+	protected $language;
+
+	/**
+	 * @var \phpbb\log\log
+	 */
+	protected $log;
+
+	/**
+	 * @var \phpbb\db\migrator
+	 */
+	protected $migrator;
+
+	/**
+	 * @var \phpbb\user
+	 */
+	protected $user;
+
+	/**
+	 * @var string
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\install\helper\container_factory				$container
+	 * @param \phpbb\filesystem\filesystem							$filesystem
+	 * @param \phpbb\install\helper\config							$installer_config
+	 * @param \phpbb\install\helper\iohandler\iohandler_interface	$iohandler
+	 * @param \phpbb\language\language								$language
+	 * @param string												$phpbb_root_path
+	 */
+	public function __construct(\phpbb\install\helper\container_factory $container, \phpbb\filesystem\filesystem $filesystem, \phpbb\install\helper\config $installer_config, \phpbb\install\helper\iohandler\iohandler_interface $iohandler, \phpbb\language\language $language, $phpbb_root_path)
+	{
+		$this->filesystem			= $filesystem;
+		$this->installer_config		= $installer_config;
+		$this->iohandler			= $iohandler;
+		$this->language				= $language;
+		$this->phpbb_root_path		= $phpbb_root_path;
+
+		$this->cache				= $container->get('cache.driver');
+		$this->config				= $container->get('config');
+		$this->extension_manager	= $container->get('ext.manager');
+		$this->log					= $container->get('log');
+		$this->migrator				= $container->get('migrator');
+		$this->user					= $container->get('user');
+
+		parent::__construct(true);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function run()
+	{
+		// @todo: Output information
+		// Log migration names and use progress bar for showing progress on migrations
+
+		$this->language->add_lang('migrator');
+
+		if (!isset($this->config['version_update_from']))
+		{
+			$this->config->set('version_update_from', $this->config['version']);
+		}
+
+		$original_version = $this->config['version_update_from'];
+
+		$this->migrator->set_output_handler(
+			new log_wrapper_migrator_output_handler(
+				$this->language,
+				new html_migrator_output_handler($this->language),
+				$this->phpbb_root_path . 'store/migrations_' . time() . '.log',
+				$this->filesystem
+			)
+		);
+
+		$this->migrator->create_migrations_table();
+
+		$migrations = $this->extension_manager
```

Comment:
Not quite sure how this question is related to this piece of code, but it would be a nice touch.